File: Assignment1/version2/Java/Driver.java

import java.util.Scanner;

/*Driver class. Takes user input for two sets
 * Afterwards asks user if they wish to perform union/intersection/difference
 * Along with the option to use the set appliction (using list)
 */
public class Driver {
	
	public static void main (String[] args){
		
		Scanner kb = new Scanner(System.in);
		
		//Get two user input and change them to sets
		System.out.println("Enter set 1 (Comma seperated, values are space and case sensitive)");
		String input1 = kb.nextLine();
		Set set1 = new Set(input1);
		
		System.out.println("Enter set 2 (Comma seperated, values are space and case sensitive)");
		String input2 = kb.nextLine();
		Set set2 = new Set(input2);
		
		int option = 0;
		
		//Loop through options until user decides to quit
		while(option != 5) {
			System.out.println("Enter action to perform: \n 1. Union \n 2. Intersection \n 3. Difference \n "
					+ "4. Application \n 5. Exit");
			try {
				option = kb.nextInt();
			}catch (Exception e) {
				System.out.println("Unknown Input");
				kb.nextLine();
				continue;
			}
			//Perform union and print result
			if (option == 1) {
				Set unionSet = set1.union(set2);
				System.out.print ("Union set: ");
				unionSet.printSet();
			}
			//Perform intersection and print result
			else if (option == 2) {
				Set intersectionSet = set1.intersection(set2);
				System.out.print ("Intersection set: ");
				intersectionSet.printSet();
			}
			//Perform difference and print result
			else if (option == 3) {
				Set differenceSet = set1.difference(set2);
				System.out.print ("Difference set: ");
				differenceSet.printSet();
			}
			//Perform application and print result
			else if (option == 4) {
				Application app = new Application();
				app.intro();
				Set a = new Set();
				Set b = new Set();
				Set c = new Set();
				
				for (int i = 0; i < app.neighbors.length; i++){
					b.add(Integer.toString(i));
				}
				app.bronKerb(a, b, c);
			}
			//End program
			else if (option == 5) {
				System.out.println("Exiting program");
				System.exit(0);
			}
			else
				System.out.println("Input is not part of choice");
		}
		
		
		kb.close();
	}

}

File: CS408 Project1/version1/C++/Driver.cpp
// CS408 Assignment 1 version 1.cpp : Defines the entry point for the console application.
//

#include <iostream>
#include "Set.h"
#include "Application.h"
#include <string>
#include <limits>
#include <exception>
#include <chrono>

using namespace std;

int main()
{
	string input1;
	string input2;

	//Get two user input for sets
	cout << "Enter set 1 (Comma seperated, values are space and case sensitive)" << endl;
	getline(cin, input1);
	cout << "Enter set 2 (Comma seperated, values are space and case sensitive)" << endl;
	getline(cin, input2);

	Set set1(input1);
	Set set2(input2);

	int option = 0;

	//Loop through options until user decides to quit
	while (option != 5) {
		cout << "Enter action to perform: \n 1. Union \n 2. Intersection \n 3. Difference \n";
		cout << " 4. Application \n 5. Exit \n";

		cin >> option;

		//Catch input mismatch
		if (cin.fail()) {
			cin.clear();
			cin.ignore(numeric_limits <streamsize>::max(), '\n');
		}
		//Perform union and print result
		if (option == 1) {
			Set unionSet = set1 + set2;
			cout << "Union set: ";
			unionSet.printSet();
		}
		//Perform intersection and print result
		else if (option == 2) {
			Set intersectionSet = set1 ^ set2;
			cout << "Intersection set: ";
			intersectionSet.printSet();
		}
		//Perform difference and print result
		else if (option == 3) {
			Set differenceSet = set1 - set2;
			cout << "Difference set: ";
			differenceSet.printSet();
		}
		//Perform application and print result
		else if (option == 4) {
			Application app;
			app.intro();
			Set a;
			Set b;
			Set c;

			for (int i = 0; i < app.getSize(); i++) {
				b.addElement(to_string(i));
			}
			
			//auto start = std::chrono::high_resolution_clock::now();

			//for (int i = 0; i < 100; i++)
				app.bronKerb(a, b, c);

			//auto finish = std::chrono::high_resolution_clock::now();

			//std::chrono::duration<double> elapsedTime = finish - start;

			//cout << "elapsed time: " << elapsedTime.count();
			//cout << "average time" << elapsedTime.count() / 100;
		}
		//End program
		else if (option == 5) {
			cout << "Exiting program";
			return 0;
		}
		else
			cout << "Input is not part of choice" << endl;
	}

    return 0;
}

